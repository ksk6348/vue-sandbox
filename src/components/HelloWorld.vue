<template>
  <div class="hello">
    <h1>{{ msg }}</h1>
    <h2>{{ message }}</h2>
    <h3>{{ value }}</h3>
  </div>
</template>

<script lang="ts">
export default {
  name: 'HelloWorld',
  props: {
    msg: String,
    val: [Number, String],
  },
  computed: {
    message(): string {
      return this.msg;
    },
    value(): string | number {
      return this.val;
    },
  },
};
</script>

<!-- Add "scoped" attribute to limit CSS to this component only -->
<style scoped>
h3 {
  margin: 40px 0 0;
}
ul {
  list-style-type: none;
  padding: 0;
}
li {
  display: inline-block;
  margin: 0 10px;
}
a {
  color: #42b983;
}
</style>
