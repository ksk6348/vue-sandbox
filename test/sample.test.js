import {mount} from '@vue/test-utils';

import App from '../src/App.vue';
import HelloWorld from '../src/components/HelloWorld.vue';

describe('test', () => {
  test('App snapshot', () => {
    const wrapper = mount(App);
    expect(wrapper.element).toMatchSnapshot();
  });

  test('HelloWorld snapshot', () => {
    const wrapper = mount(HelloWorld, {
      propsData: {
        msg: 'test',
        val: 123,
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });
});
