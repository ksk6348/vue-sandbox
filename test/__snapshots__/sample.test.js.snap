// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test App snapshot 1`] = `
<div
  id="app"
>
  <img
    alt="Vue logo"
    src="./assets/logo.png"
  />
   
  <div
    class="hello"
  >
    <h1>
      Welcome to Your Vue.js App
    </h1>
     
    <h2>
      Welcome to Your Vue.js App
    </h2>
     
    <h3>
      123
    </h3>
  </div>
</div>
`;

exports[`test HelloWorld snapshot 1`] = `
<div
  class="hello"
>
  <h1>
    test
  </h1>
   
  <h2>
    test
  </h2>
   
  <h3>
    123
  </h3>
</div>
`;
